examples/run/run.cpp

```diff
@@ -573,21 +590,76 @@ class LlamaData {
     }
 #endif
 
-    int huggingface_dl(const std::string & model, const std::vector<std::string> headers, const std::string & bn) {
+    int huggingface_dl_tmpl(const std::string & hfr, const std::vector<std::string> headers, const std::string & tn) {
+        // if template already exists, don't download it
+        struct stat info;
+        if (stat(tn.c_str(), &info) == 0) {
```

Comment:
Not a strictly required change, but we use std::filesystem::exists everywhere else in llama-run to checkout for existence of files. Does stat work on Windows?